src/main/java/uk/gov/hmcts/reform/et/syaapi/controllers/CaseRoleManagementController.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.reform.et.syaapi.controllers;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseAssignmentUserRolesRequest;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.et.syaapi.annotation.ApiResponseGroup;
+import uk.gov.hmcts.reform.et.syaapi.exception.CaseRoleManagementException;
+import uk.gov.hmcts.reform.et.syaapi.models.FindCaseForRoleModificationRequest;
+import uk.gov.hmcts.reform.et.syaapi.models.NotifyUserCaseRoleModificationRequest;
+import uk.gov.hmcts.reform.et.syaapi.service.CaseRoleManagementService;
+
+import javax.validation.constraints.NotNull;
+
+import static org.springframework.http.ResponseEntity.ok;
+import static uk.gov.hmcts.reform.et.syaapi.constants.CaseRoleManagementConstants.CASE_USER_ROLE_SUCCESSFULLY_MODIFIED;
+import static uk.gov.hmcts.reform.et.syaapi.constants.EtSyaConstants.AUTHORIZATION;
+
+/**
+ * Rest Controller to modify case user roles.
+ */
+@Slf4j
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/caseRoleManagement")
+public class CaseRoleManagementController {
+
+    private final CaseRoleManagementService caseRoleManagementService;
+
+    @PostMapping("/findCaseForRoleModification")
+    @Operation(summary = "Modifies user roles of the case")
+    @ApiResponseGroup
+    public ResponseEntity<CaseDetails> findCaseForRoleModification(
+        @NotNull @RequestBody FindCaseForRoleModificationRequest findCaseForRoleModificationRequest
+    ) {
+        CaseDetails caseDetails =
+            caseRoleManagementService.findCaseForRoleModification(findCaseForRoleModificationRequest);
+        return ok(caseDetails);
```

Comment:
Given that null can be returned from this, do we want to handle this differently?

Reply:
Handling it in the frontend as if no case details returns throws an error. It will allow me to add an option to forward to a new error page instead of having thrown error shown in the frontend.
const caseData = (await getCaseApi(req.session.user?.accessToken)?.getCaseByApplicationRequest(req))?.data;
      if (caseData) {
        req.session.userCase = formatApiCaseDataToCaseWithId(caseData);
        SelfAssignmentFormControllerHelper.setRespondentName(req, caseData);
        return res.redirect(setUrlLanguage(req, PageUrls.SELF_ASSIGNMENT_CHECK));
      }
      ErrorUtils.setManualErrorToRequestSession(
        req,
        ValidationErrors.API,
        FormFieldNames.SELF_ASSIGNMENT_FORM_FIELDS.HIDDEN_ERROR_FIELD
      );
      return res.redirect(req.url);
    } else {
      req.session.errors = errors;
      req.session.save(err => {
        if (err) {
          throw err;
        }
        return res.redirect(req.url);
      });
    }

Reply:
It is being handled in the frontend to show a new api validation error in the frontend. When no data returns adds an error to a hidden field and shows that the input values doesn't match with any case as in the image. 
const caseData = (await getCaseApi(req.session.user?.accessToken)?.getCaseByApplicationRequest(req))?.data;
      if (caseData) {
        req.session.userCase = formatApiCaseDataToCaseWithId(caseData);
        SelfAssignmentFormControllerHelper.setRespondentName(req, caseData);
        return res.redirect(setUrlLanguage(req, PageUrls.SELF_ASSIGNMENT_CHECK));
      }
      ErrorUtils.setManualErrorToRequestSession(
        req,
        ValidationErrors.API,
        FormFieldNames.SELF_ASSIGNMENT_FORM_FIELDS.HIDDEN_ERROR_FIELD
      );
![localhost_3003_self-assignment-form_isNew=true](https://github.com/user-attachments/assets/5f943c56-ff92-4b55-b124-a14833d0bc72)